### PR TITLE
feat: add impending deletion banner to workspace page

### LINKED ISSUE #7359: workspace actions: add impending deletion banner to the workspace page
- add a persistent banner to the workspace page that appears when a workspace is scheduled for deletion within 14 days 
 - we should add a boolean in local storage so that this toast only needs to be dismissed once per session

### PR DESCRIPTION
**Pull request by Kira-Pilot**
Screenshot 2023-05-22 at 3 06 01 PM

resolves #7359

### CODE REVIEW

**Comment by Kira-Pilot** [comment]
@kylecarbs, @bpmct and I would like to know how you feel about this copy. It will probably change a bit when we add a workspace archival component to this feature, but curious if you have initial thoughts.

**Comment by ammario** [comment]
image
This banner should be a lot louder IMO, maybe when days left is <7? Maybe this orphan banner can provide inspiration:

image


The copy looks good to me.

**Comment by Kira-Pilot** [comment]
Screenshot 2023-05-24 at 12 54 22 PM

@ammario I'm hesitant to introduce a new UI pattern, i.e. a new banner style; however I did add some logic to show our 'error' banner flavor when workspaces will be deleted in 7 days or less so as to convey more urgency. This banner will also be displayed on the `/workspaces` page and we'll have additional badges and VS Code notifications.

File: site/src/components/WorkspaceDeletion/ImpendingDeletionBanner.tsx [hunk 1]
```diff
@@ -20,21 +27,35 @@ export const ImpendingDeletionBanner = ({
...
+    !workspace ||
+    !displayImpendingDeletion(
+      workspace,
+      allowAdvancedScheduling,
+      allowWorkspaceActions,
+    ) ||
+    !displayImpendingDeletionBanner
+  ) {
+    return null
+  }
```
Comment (BrunoQuaresma) [inline site/src/components/WorkspaceDeletion/ImpendingDeletionBanner.tsx:40]: I found it hard to read/understand this "display or not" logic here. I think the names `displayImpendingDeletion` and `displayImpendingDeletionBanner` are very similar and I took some time to understand they are different but not sure how to improve this.

**Review by BrunoQuaresma — APPROVED** [review]

**Comment by BrunoQuaresma** [comment]
One comment about this feature overall, maybe would be nice to add a help popover in the "Impending deletion" status explaining what it means.

File: site/src/components/WorkspaceDeletion/ImpendingDeletionBanner.tsx (hunk 1 above)
Comment (Kira-Pilot) [reply site/src/components/WorkspaceDeletion/ImpendingDeletionBanner.tsx:40]: Ah, let me improve that

**Pull request merged**